### ISSUE TITLE
Cannot pass formatter to deal with CircularJSON

### ISSUE THREAD

**Issue by aqueiros** [pkallos/winston-firehose#22]
Greetings,

I am having a problem with circular json, much like what is reported in winston:
https://github.com/winstonjs/winston/issues/1348

except I am using firehose winston and I am unable to specify a formatter to use. I am trying:

```
new WFirehose({
      streamName: process.env.LOGGER_STREAM_NAME,
      firehoseOptions: {
        region: process.env.AWS_REGION
      },
      formatter: winston.format.json
    })
```

And getting 
```
UnhandledPromiseRejectionWarning: InvalidParameterType: Expected params.Record.Data to be a string, Buffer, Stream, Blob, or typed array object
    at ParamValidator.fail
```

I also tried using json-stringify-safe, circular-json and fast-safe-stringify ending up with

```
UnknownError: 413
```

Can someone help? From looking at the code, it should be possible to replace the formatter:

```
const FirehoseLogger = class FirehoseLogger extends Transport {
  constructor(options) {
    super(options);
    this.name = 'FirehoseLogger';
    this.level = options.level || 'info';
    this.formatter = options.formatter || JSON.stringify; <------
```

**Comment by pkallos** [comment]
This might be resolved in `2.1.0`, fixed by #21 .

**Comment by pkallos** [comment]
@aqueiros are you seeing this as resolved in versions `>2.1.0`? closing for now.
